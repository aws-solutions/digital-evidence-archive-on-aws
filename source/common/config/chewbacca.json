{
  "stage": "chewbacca",
  "testStack": true,
  "userGroups": [
    {
      "name": "CaseWorkerGroup",
      "description": "containing users who need access to case APIs",
      "precedence": 1,
      "endpoints": [
        {
          "path": "/cases",
          "method": "GET"
        },
        {
          "path": "/cases",
          "method": "POST"
        },
        {
          "path": "/cases/{caseId}",
          "method": "GET"
        },
        {
          "path": "/cases/{caseId}",
          "method": "PUT"
        },
        {
          "path": "/cases/{caseId}",
          "method": "DELETE"
        },
        {
          "path": "/cases/{caseId}/userMemberships",
          "method": "POST"
        }
      ]
    },
    {
      "name": "AuthTestGroup",
      "description": "used for auth e2e testing",
      "precedence": 100,
      "endpoints": [
        {
          "path": "/hi",
          "method": "GET"
        },
        {
          "path": "/bye",
          "method": "GET"
        }
      ]
    },
    {
      "name": "CreateCasesTestGroup",
      "description": "used for create cases API e2e testing",
      "precedence": 100,
      "endpoints": [
        {
          "path": "/cases",
          "method": "POST"
        },
        {
          "path": "/cases/{caseId}",
          "method": "DELETE"
        },
        {
          "path": "/cases/all-cases",
          "method": "GET"
        }
      ]
    },
    {
      "name": "GetCaseTestGroup",
      "description": "used for get cases API e2e testing",
      "precedence": 100,
      "endpoints": [
        {
          "path": "/cases",
          "method": "POST"
        },
        {
          "path": "/cases/{caseId}",
          "method": "GET"
        },
        {
          "path": "/cases/{caseId}",
          "method": "DELETE"
        }
      ]
    },
    {
      "name": "GetMyCasesTestGroup",
      "description": "used for get my cases API e2e testing",
      "precedence": 100,
      "endpoints": [
        {
          "path": "/cases",
          "method": "POST"
        },
        {
          "path": "/cases/{caseId}",
          "method": "GET"
        },
        {
          "path": "/cases/{caseId}",
          "method": "DELETE"
        },
        {
          "path": "/cases/my-cases",
          "method": "GET"
        },
        {
          "path": "/cases/{caseId}/userMemberships",
          "method": "POST"
        }
      ]
    }
  ]
}
