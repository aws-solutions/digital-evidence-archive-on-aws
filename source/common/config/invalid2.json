{
  "deaRoleTypes": [
    {
      "name": "InvalidEndpointConfig",
      "description": "used for auth e2e testing",
      "endpoints": "InvalidEndpoints"
    }
  ]
}
