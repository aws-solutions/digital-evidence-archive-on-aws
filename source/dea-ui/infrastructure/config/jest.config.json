{
  "extends": "@rushstack/heft-node-rig/profiles/default/config/jest.config.json",
  "collectCoverage": true,
  "coveragePathIgnorePatterns": ["<rootDir>/src/index.ts"],
  "coverageThreshold": {
    "global": {
      "branches": 75,
      "functions": 75,
      "lines": 75,
      "statements": 75
    }
  },
  "coverageReporters": ["json-summary", "json", "text"]
}
