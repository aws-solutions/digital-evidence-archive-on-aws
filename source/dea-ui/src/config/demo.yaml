# Stage Name
stage: demo
awsRegion: us-west-2
awsRegionShortName: sea

apiUrlOutput: 'https://dea.demo'
stacks:
  - name: teststack
    WebsiteURL: 'https://ui.stack'
