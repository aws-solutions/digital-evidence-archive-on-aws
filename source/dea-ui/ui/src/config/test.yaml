awsRegionShortName: 'Testville'
awsRegion: 'us-test-1'
apiUrlOutput: 'https://bogus.bogus'
stacks:
  - name: teststack
    WebsiteURL: 'https://bogus.bogus'
