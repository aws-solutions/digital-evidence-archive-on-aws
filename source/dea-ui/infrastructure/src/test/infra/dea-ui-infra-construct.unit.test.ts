/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import { addSnapshotSerializers } from '@aws/dea-backend';
import { convictConfig } from '@aws/dea-backend/lib/config';
import * as cdk from 'aws-cdk-lib';
import { Duration, RemovalPolicy, Stack } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { RestApi } from 'aws-cdk-lib/aws-apigateway';
import { Key } from 'aws-cdk-lib/aws-kms';
import { BlockPublicAccess, Bucket, ObjectOwnership } from 'aws-cdk-lib/aws-s3';
import 'source-map-support/register';
import { validateDeaUiConstruct } from '../..';
import { DeaUiConstruct } from '../../dea-ui-stack';

describe('DEA UI Infrastructure stack', () => {
  beforeAll(() => {
    process.env.STAGE = 'devsample';
    process.env.DIST_OUTPUT_BUCKET = 'DOC-EXAMPLE-BUCKET';
  });

  afterAll(() => {
    delete process.env.STAGE;
  });

  it('synthesizes the way we expect', () => {
    const app = new cdk.App();
    const stack = new Stack(app, 'test-stack');

    const key = new Key(stack, 'testKey', {
      enableKeyRotation: true,
      removalPolicy: RemovalPolicy.DESTROY,
      pendingWindow: Duration.days(7),
    });

    const accessLogsBucket = new Bucket(stack, 'testS3AccessLogBucket', {
      blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
      objectOwnership: ObjectOwnership.BUCKET_OWNER_PREFERRED,
    });

    const restApi = new RestApi(stack, 'testApi', { description: 'Backend API' });

    new DeaUiConstruct(stack, 'DeaUiConstruct', {
      kmsKey: key,
      accessLogsBucket: accessLogsBucket,
      restApi,
      accessLogPrefix: 'dea-ui-access-log',
    });

    // Prepare the stack for assertions.
    const template = Template.fromStack(stack);
    // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/consistent-type-assertions
    delete template['template']['Mappings'];

    // assertions relevant to backend and any parent
    validateDeaUiConstruct(template);

    // ui-specific assertions
    template.allResourcesProperties('AWS::S3::Bucket', {
      PublicAccessBlockConfiguration: {
        BlockPublicAcls: true,
        BlockPublicPolicy: true,
        IgnorePublicAcls: true,
        RestrictPublicBuckets: true,
      },
    });

    //handlers + authorizer
    const expectedBucketCount = 2;
    const expectedLambdaCount = 2;
    const expectedMethodCount = 7;
    template.resourceCountIs('AWS::S3::Bucket', expectedBucketCount);
    template.resourceCountIs('AWS::ApiGateway::Method', expectedMethodCount);
    template.resourceCountIs('AWS::Lambda::Function', expectedLambdaCount);

    addSnapshotSerializers();

    expect(template).toMatchSnapshot();
  });

  it('synthesizes the with isOneClick flag enabled', () => {
    // set isOneClick to true
    convictConfig.set('isOneClick', true);

    const app = new cdk.App();
    const stack = new Stack(app, 'test-stack');

    const key = new Key(stack, 'testKey', {
      enableKeyRotation: true,
      removalPolicy: RemovalPolicy.DESTROY,
      pendingWindow: Duration.days(7),
    });

    const accessLogsBucket = new Bucket(stack, 'testS3AccessLogBucket', {
      blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
      objectOwnership: ObjectOwnership.BUCKET_OWNER_PREFERRED,
    });

    const restApi = new RestApi(stack, 'testApi', { description: 'Backend API' });

    new DeaUiConstruct(stack, 'DeaUiConstruct', {
      kmsKey: key,
      accessLogsBucket: accessLogsBucket,
      restApi,
      accessLogPrefix: 'dea-ui-access-log',
    });

    // Prepare the stack for assertions.
    const template = Template.fromStack(stack);

    // assertions relevant to backend and any parent
    validateDeaUiConstruct(template);

    // ui-specific assertions
    template.allResourcesProperties('AWS::S3::Bucket', {
      PublicAccessBlockConfiguration: {
        BlockPublicAcls: true,
        BlockPublicPolicy: true,
        IgnorePublicAcls: true,
        RestrictPublicBuckets: true,
      },
    });

    //handlers + authorizer
    const expectedBucketCount = 2;
    const expectedLambdaCount = 2;
    const expectedMethodCount = 7;
    template.resourceCountIs('AWS::S3::Bucket', expectedBucketCount);
    template.resourceCountIs('AWS::ApiGateway::Method', expectedMethodCount);
    template.resourceCountIs('AWS::Lambda::Function', expectedLambdaCount);
  });
});
