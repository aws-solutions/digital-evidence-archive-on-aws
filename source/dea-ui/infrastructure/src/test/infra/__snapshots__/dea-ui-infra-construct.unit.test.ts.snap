// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeaMainStack synthesizes the way we expect 1`] = `
Object {
  "Outputs": Object {
    "testApiEndpointC9D07ADA": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Ref": "testApiD6ECAB50",
            },
            ".execute-api.",
            Object {
              "Ref": "AWS::Region",
            },
            ".",
            Object {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            Object {
              "Ref": "testApiDeploymentStageprodCE051BE8",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536": Object {
      "DependsOn": Array [
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "[HASH REMOVED].zip",
        },
        "Handler": "index.handler",
        "Layers": Array [
          Object {
            "Ref": "DeaUiStackDeaUiStackdeploymentbucketAwsCliLayerE6B55D5A",
          },
        ],
        "Role": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
            "Arn",
          ],
        },
        "Runtime": "python3.9",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "DeaUiStackDeaUiStackbucket2409F549",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "DeaUiStackDeaUiStackbucket2409F549",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "Roles": Array [
          Object {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F": Object {
      "DependsOn": Array [
        "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "[HASH REMOVED].zip",
        },
        "Description": Object {
          "Fn::Join": Array [
            "",
            Array [
              "Lambda function for auto-deleting objects in ",
              Object {
                "Ref": "DeaUiStackDeaUiStackbucket2409F549",
              },
              " S3 bucket.",
            ],
          ],
        },
        "Handler": "__entrypoint__.handler",
        "MemorySize": 128,
        "Role": Object {
          "Fn::GetAtt": Array [
            "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "DeaUiStackDeaUiStackbucket2409F549": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "AccessControl": "LogDeliveryWrite",
        "BucketEncryption": Object {
          "ServerSideEncryptionConfiguration": Array [
            Object {
              "ServerSideEncryptionByDefault": Object {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "LoggingConfiguration": Object {
          "DestinationBucketName": Object {
            "Ref": "testS3AccessLogBucketF0DCA384",
          },
          "LogFilePrefix": "dea-ui-access-log",
        },
        "PublicAccessBlockConfiguration": Object {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "Tags": Array [
          Object {
            "Key": "aws-cdk:auto-delete-objects",
            "Value": "true",
          },
          Object {
            "Key": "aws-cdk:cr-owned:fb025039",
            "Value": "true",
          },
        ],
        "WebsiteConfiguration": Object {
          "IndexDocument": "index.html",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "DeaUiStackDeaUiStackbucketAutoDeleteObjectsCustomResource59A9A4D8": Object {
      "DeletionPolicy": "Delete",
      "DependsOn": Array [
        "DeaUiStackDeaUiStackbucketPolicy3F40F0AF",
      ],
      "Properties": Object {
        "BucketName": Object {
          "Ref": "DeaUiStackDeaUiStackbucket2409F549",
        },
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F",
            "Arn",
          ],
        },
      },
      "Type": "Custom::S3AutoDeleteObjects",
      "UpdateReplacePolicy": "Delete",
    },
    "DeaUiStackDeaUiStackbucketPolicy3F40F0AF": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "DeaUiStackDeaUiStackbucket2409F549",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
                    "Arn",
                  ],
                },
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "DeaUiStackDeaUiStackbucket2409F549",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "DeaUiStackDeaUiStackbucket2409F549",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": "s3:*",
              "Condition": Object {
                "Bool": Object {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": Object {
                "AWS": "*",
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "DeaUiStackDeaUiStackbucket2409F549",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "DeaUiStackDeaUiStackbucket2409F549",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
              "Sid": "Deny requests that do not use TLS/HTTPS",
            },
            Object {
              "Action": "s3:*",
              "Condition": Object {
                "StringNotEquals": Object {
                  "s3:signatureversion": "AWS4-HMAC-SHA256",
                },
              },
              "Effect": "Deny",
              "Principal": Object {
                "AWS": "*",
              },
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "DeaUiStackDeaUiStackbucket2409F549",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
              "Sid": "Deny requests that do not use SigV4",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "DeaUiStackDeaUiStackdeploymentbucketAwsCliLayerE6B55D5A": Object {
      "Properties": Object {
        "Content": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "[HASH REMOVED].zip",
        },
        "Description": "/opt/awscli/aws",
      },
      "Type": "AWS::Lambda::LayerVersion",
    },
    "DeaUiStackDeaUiStackdeploymentbucketCustomResource5769B150": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "DestinationBucketName": Object {
          "Ref": "DeaUiStackDeaUiStackbucket2409F549",
        },
        "Prune": true,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536",
            "Arn",
          ],
        },
        "SourceBucketNames": Array [
          Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
        ],
        "SourceObjectKeys": Array [
          "[HASH REMOVED].zip",
        ],
      },
      "Type": "Custom::CDKBucketDeployment",
      "UpdateReplacePolicy": "Delete",
    },
    "DeaUiStackrole3432A944": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "DeaUiStackroleDefaultPolicyBE7B3383": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "DeaUiStackDeaUiStackbucket2409F549",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "DeaUiStackDeaUiStackbucket2409F549",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "DeaUiStackroleDefaultPolicyBE7B3383",
        "Roles": Array [
          Object {
            "Ref": "DeaUiStackrole3432A944",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "testApiAccountA6723CBB": Object {
      "DeletionPolicy": "Retain",
      "DependsOn": Array [
        "testApiD6ECAB50",
      ],
      "Properties": Object {
        "CloudWatchRoleArn": Object {
          "Fn::GetAtt": Array [
            "testApiCloudWatchRole8A11639C",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
      "UpdateReplacePolicy": "Retain",
    },
    "testApiCloudWatchRole8A11639C": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
      "UpdateReplacePolicy": "Retain",
    },
    "testApiD6ECAB50": Object {
      "Properties": Object {
        "Description": "Backend API",
        "Name": "testApi",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "testApiDeployment3727A0B911a3d6f3089853cb0ac40dd95a583e66": Object {
      "DependsOn": Array [
        "testApiuiproxyGET92230E90",
        "testApiuiproxyDDF9EC1E",
        "testApiuiGET8FD5D6F5",
        "testApiui1F041DD2",
      ],
      "Properties": Object {
        "Description": "Backend API",
        "RestApiId": Object {
          "Ref": "testApiD6ECAB50",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "testApiDeploymentStageprodCE051BE8": Object {
      "DependsOn": Array [
        "testApiAccountA6723CBB",
      ],
      "Properties": Object {
        "DeploymentId": Object {
          "Ref": "testApiDeployment3727A0B911a3d6f3089853cb0ac40dd95a583e66",
        },
        "RestApiId": Object {
          "Ref": "testApiD6ECAB50",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "testApiui1F041DD2": Object {
      "Properties": Object {
        "ParentId": Object {
          "Fn::GetAtt": Array [
            "testApiD6ECAB50",
            "RootResourceId",
          ],
        },
        "PathPart": "ui",
        "RestApiId": Object {
          "Ref": "testApiD6ECAB50",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "testApiuiGET8FD5D6F5": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "GET",
        "Integration": Object {
          "ContentHandling": "CONVERT_TO_TEXT",
          "Credentials": Object {
            "Fn::GetAtt": Array [
              "DeaUiStackrole3432A944",
              "Arn",
            ],
          },
          "IntegrationHttpMethod": "GET",
          "IntegrationResponses": Array [
            Object {
              "ResponseParameters": Object {
                "method.response.header.Content-Type": "integration.response.header.Content-Type",
              },
              "StatusCode": "200",
            },
          ],
          "PassthroughBehavior": "WHEN_NO_TEMPLATES",
          "RequestParameters": Object {
            "integration.request.header.Accept": "method.request.header.Accept",
            "integration.request.header.Content-Type": "method.request.header.Content-Type",
            "integration.request.path.proxy": "method.request.path.proxy",
          },
          "Type": "AWS",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":s3:path/",
                Object {
                  "Ref": "DeaUiStackDeaUiStackbucket2409F549",
                },
                "/index.html",
              ],
            ],
          },
        },
        "MethodResponses": Array [
          Object {
            "ResponseModels": Object {
              "application/json": "Empty",
            },
            "ResponseParameters": Object {
              "method.response.header.Content-Length": true,
              "method.response.header.Content-Type": true,
            },
            "StatusCode": "200",
          },
        ],
        "RequestParameters": Object {
          "method.request.header.Accept": true,
          "method.request.header.Content-Type": true,
          "method.request.path.proxy": true,
        },
        "ResourceId": Object {
          "Ref": "testApiui1F041DD2",
        },
        "RestApiId": Object {
          "Ref": "testApiD6ECAB50",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "testApiuiproxyDDF9EC1E": Object {
      "Properties": Object {
        "ParentId": Object {
          "Ref": "testApiui1F041DD2",
        },
        "PathPart": "{proxy+}",
        "RestApiId": Object {
          "Ref": "testApiD6ECAB50",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "testApiuiproxyGET92230E90": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "GET",
        "Integration": Object {
          "ContentHandling": "CONVERT_TO_TEXT",
          "Credentials": Object {
            "Fn::GetAtt": Array [
              "DeaUiStackrole3432A944",
              "Arn",
            ],
          },
          "IntegrationHttpMethod": "GET",
          "IntegrationResponses": Array [
            Object {
              "ResponseParameters": Object {
                "method.response.header.Content-Type": "integration.response.header.Content-Type",
              },
              "StatusCode": "200",
            },
          ],
          "PassthroughBehavior": "WHEN_NO_TEMPLATES",
          "RequestParameters": Object {
            "integration.request.header.Accept": "method.request.header.Accept",
            "integration.request.header.Content-Type": "method.request.header.Content-Type",
            "integration.request.path.proxy": "method.request.path.proxy",
          },
          "Type": "AWS",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":s3:path/",
                Object {
                  "Ref": "DeaUiStackDeaUiStackbucket2409F549",
                },
                "/{proxy}",
              ],
            ],
          },
        },
        "MethodResponses": Array [
          Object {
            "ResponseModels": Object {
              "application/json": "Empty",
            },
            "ResponseParameters": Object {
              "method.response.header.Content-Length": true,
              "method.response.header.Content-Type": true,
            },
            "StatusCode": "200",
          },
        ],
        "RequestParameters": Object {
          "method.request.header.Accept": true,
          "method.request.header.Content-Type": true,
          "method.request.path.proxy": true,
        },
        "ResourceId": Object {
          "Ref": "testApiuiproxyDDF9EC1E",
        },
        "RestApiId": Object {
          "Ref": "testApiD6ECAB50",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "testKey1CDDDD5E": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "EnableKeyRotation": true,
        "KeyPolicy": Object {
          "Statement": Array [
            Object {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PendingWindowInDays": 7,
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Delete",
    },
    "testS3AccessLogBucketF0DCA384": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "AccessControl": "LogDeliveryWrite",
        "PublicAccessBlockConfiguration": Object {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
